### ISSUE TITLE
Querying failure after scaling job success on existing schema since tableMetaData lost in ShardingSphereSchema

### ISSUE THREAD

**Issue by sandynz** [apache/shardingsphere#16207]
### Which version of ShardingSphere did you use?
master branch, commit b7d3205fbbd634ffbaf2abd6d1c5318743d9cbfc

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy, scaling enabled

### Expected behavior
Query works after scaling job success.

### Actual behavior
There's an existing schema, alter sharding table rule to emit scaling job.

Execute select query after scaling job success. But it failed:
```
mysql> show scaling status 0130317c30317c3154327c7363616c696e675f646232;
+------+-------------+----------+--------+-------------------------------+--------------------------+
| item | data_source | status   | active | inventory_finished_percentage | incremental_idle_seconds |
+------+-------------+----------+--------+-------------------------------+--------------------------+
| 0    | ds_0        | FINISHED | false  | 100                           | 157                      |
| 1    | ds_1        | FINISHED | false  | 100                           | 157                      |
+------+-------------+----------+--------+-------------------------------+--------------------------+
2 rows in set (0.02 sec)

mysql> preview select count(1) from t_order;
+------------------+-------------------------------------------------------------------------+
| data_source_name | actual_sql                                                              |
+------------------+-------------------------------------------------------------------------+
| ds_2             | select count(1) from t_order_0 UNION ALL select count(1) from t_order_3 |
| ds_3             | select count(1) from t_order_1 UNION ALL select count(1) from t_order_4 |
| ds_4             | select count(1) from t_order_2 UNION ALL select count(1) from t_order_5 |
+------------------+-------------------------------------------------------------------------+
3 rows in set (0.00 sec)

mysql> select count(1) from t_order;
ERROR 1997 (C1997): Runtime exception: [null]
```
Exception log:
```
[ERROR] 2022-03-19 16:56:14.458 [ShardingSphere-Command-1] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
java.lang.NullPointerException: null
	at org.apache.shardingsphere.sharding.merge.dql.groupby.GroupByMemoryMergedResult.getValueCaseSensitiveFromTables(GroupByMemoryMergedResult.java:134)
	at org.apache.shardingsphere.sharding.merge.dql.groupby.GroupByMemoryMergedResult.getValueCaseSensitive(GroupByMemoryMergedResult.java:124)
	at org.apache.shardingsphere.sharding.merge.dql.groupby.GroupByMemoryMergedResult.init(GroupByMemoryMergedResult.java:73)
	at org.apache.shardingsphere.sharding.merge.dql.groupby.GroupByMemoryMergedResult.init(GroupByMemoryMergedResult.java:53)
	at org.apache.shardingsphere.infra.merge.result.impl.memory.MemoryMergedResult.<init>(MemoryMergedResult.java:51)
	at org.apache.shardingsphere.sharding.merge.dql.groupby.GroupByMemoryMergedResult.<init>(GroupByMemoryMergedResult.java:56)
	at org.apache.shardingsphere.sharding.merge.dql.ShardingDQLResultMerger.getGroupByMergedResult(ShardingDQLResultMerger.java:112)
	at org.apache.shardingsphere.sharding.merge.dql.ShardingDQLResultMerger.build(ShardingDQLResultMerger.java:82)
	at org.apache.shardingsphere.sharding.merge.dql.ShardingDQLResultMerger.merge(ShardingDQLResultMerger.java:63)
	at org.apache.shardingsphere.infra.merge.MergeEngine.executeMerge(MergeEngine.java:90)
	at org.apache.shardingsphere.infra.merge.MergeEngine.merge(MergeEngine.java:80)
	at org.apache.shardingsphere.proxy.backend.communication.DatabaseCommunicationEngine.mergeQuery(DatabaseCommunicationEngine.java:159)
	at org.apache.shardingsphere.proxy.backend.communication.DatabaseCommunicationEngine.processExecuteQuery(DatabaseCommunicationEngine.java:113)
	at org.apache.shardingsphere.proxy.backend.communication.jdbc.JDBCDatabaseCommunicationEngine.execute(JDBCDatabaseCommunicationEngine.java:134)
	at org.apache.shardingsphere.proxy.backend.communication.jdbc.JDBCDatabaseCommunicationEngine.execute(JDBCDatabaseCommunicationEngine.java:70)
	at org.apache.shardingsphere.proxy.backend.text.data.impl.SchemaAssignedDatabaseBackendHandler.execute(SchemaAssignedDatabaseBackendHandler.java:55)
	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.text.query.MySQLComQueryPacketExecutor.execute(MySQLComQueryPacketExecutor.java:67)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:97)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:69)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

### Reason analyze (If you can)
TableMetaData map `tables` lose logic tables metadata in ShardingSphereSchema after versioned configuration switched when scaling job success.

<img width="1401" alt="WX20220319-172945@2x" src="https://user-images.githubusercontent.com/42492540/159116182-aa9c5dad-2c48-4eab-9e99-3215d703436e.png">

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
Except executing scaling job, the key is switching metadata, so we could just switch metadata manually to reproduct this issue.

Make sure `select count(1) from t_order;` works on current rules.
Prepare another database and sharded tables, add new resources via DistSQL or in ZK client, write related sharding rule and just replace rules in ZK client. Then execute `select count(1) from t_order;` again.

**Comment by sandynz** [comment]
I tested more scenarios, they might be caused by the same bug, so I put these testing here too.

On master branch, commit 660449bba6bf89217a0ea4a4510ce6c4a4dd9962 , and need to revert commit 8f012fe4adff996fd796feec038d3837d4a704cc for ZK.watch.

Reproduce steps:
```
mysql> create database test1;
Query OK, 0 rows affected (0.05 sec)

mysql> use test1
Database changed
mysql> ADD RESOURCE ds_0 (
    ->     URL="jdbc:mysql://127.0.0.1:3306/scaling_ds_0?serverTimezone=UTC&useSSL=false",
    ->     USER=root,
    ->     PASSWORD=root,
    ->     PROPERTIES("maximumPoolSize"=50,"idleTimeout"="60000")
    -> ), ds_1 (
    ->     URL="jdbc:mysql://127.0.0.1:3306/scaling_ds_1?serverTimezone=UTC&useSSL=false",
    ->     USER=root,
    ->     PASSWORD=root,
    ->     PROPERTIES("maximumPoolSize"=50,"idleTimeout"="60000")
    -> );
Query OK, 0 rows affected (0.29 sec)

mysql>
mysql> ADD RESOURCE ds_2 (
    ->     URL="jdbc:mysql://127.0.0.1:3306/scaling_ds_10?serverTimezone=UTC&useSSL=false",
    ->     USER=root,
    ->     PASSWORD=root,
    ->     PROPERTIES("maximumPoolSize"=50,"idleTimeout"="60000")
    -> ), ds_3 (
    ->     URL="jdbc:mysql://127.0.0.1:3306/scaling_ds_11?serverTimezone=UTC&useSSL=false",
    ->     USER=root,
    ->     PASSWORD=root,
    ->     PROPERTIES("maximumPoolSize"=50,"idleTimeout"="60000")
    -> ), ds_4 (
    ->     URL="jdbc:mysql://127.0.0.1:3306/scaling_ds_12?serverTimezone=UTC&useSSL=false",
    ->     USER=root,
    ->     PASSWORD=root,
    ->     PROPERTIES("maximumPoolSize"=50,"idleTimeout"="60000")
    -> );
Query OK, 0 rows affected (0.24 sec)

mysql> CREATE SHARDING TABLE RULE t_order(
    -> RESOURCES(ds_0,ds_1),
    -> SHARDING_COLUMN=order_id,
    -> TYPE(NAME=hash_mod,PROPERTIES("sharding-count"=4)),
    -> KEY_GENERATE_STRATEGY(COLUMN=order_id,TYPE(NAME=snowflake,PROPERTIES("worker-id"=123)))
    -> ), t_order_item(
    -> RESOURCES(ds_0,ds_1),
    -> SHARDING_COLUMN=order_id,
    -> TYPE(NAME=hash_mod,PROPERTIES("sharding-count"=4)),
    -> KEY_GENERATE_STRATEGY(COLUMN=order_id,TYPE(NAME=snowflake,PROPERTIES("worker-id"=123)))
    -> );
Query OK, 0 rows affected (1.21 sec)

mysql>
mysql> CREATE SHARDING BINDING TABLE RULES (t_order,t_order_item);
Query OK, 0 rows affected (0.04 sec)

mysql> CREATE SHARDING SCALING RULE scaling_name1(
    -> COMPLETION_DETECTOR(TYPE(NAME=IDLE, PROPERTIES("incremental-task-idle-second-threshold"=1))),
    -> DATA_CONSISTENCY_CHECKER(TYPE(NAME=DATA_MATCH, PROPERTIES("chunk-size"=1000)))
    -> );
Query OK, 0 rows affected (0.05 sec)

mysql> preview select count(1) from t_order;
+------------------+-------------------------------------------------------------------------+
| data_source_name | actual_sql                                                              |
+------------------+-------------------------------------------------------------------------+
| ds_0             | select count(1) from t_order_0 UNION ALL select count(1) from t_order_2 |
| ds_1             | select count(1) from t_order_1 UNION ALL select count(1) from t_order_3 |
+------------------+-------------------------------------------------------------------------+
2 rows in set (0.37 sec)

mysql> select count(1) from t_order;
ERROR 1997 (C1997): Runtime exception: [null]

mysql> ALTER SHARDING TABLE RULE t_order(
    -> RESOURCES(ds_2,ds_3,ds_4),
    -> SHARDING_COLUMN=order_id,
    -> TYPE(NAME=hash_mod,PROPERTIES("sharding-count"=6)),
    -> KEY_GENERATE_STRATEGY(COLUMN=order_id,TYPE(NAME=snowflake,PROPERTIES("worker-id"=123)))
    -> ), t_order_item(
    -> RESOURCES(ds_2,ds_3,ds_4),
    -> SHARDING_COLUMN=order_id,
    -> TYPE(NAME=hash_mod,PROPERTIES("sharding-count"=6)),
    -> KEY_GENERATE_STRATEGY(COLUMN=order_id,TYPE(NAME=snowflake,PROPERTIES("worker-id"=123)))
    -> );
Query OK, 0 rows affected (0.27 sec)

mysql> show scaling status 0130317c30317c3054317c7465737431;
+------+-------------+--------------------------+--------+-------------------------------+--------------------------+
| item | data_source | status                   | active | inventory_finished_percentage | incremental_idle_seconds |
+------+-------------+--------------------------+--------+-------------------------------+--------------------------+
| 0    | ds_0        | EXECUTE_INCREMENTAL_TASK | true   | 100                           | 474                      |
| 1    | ds_1        | EXECUTE_INCREMENTAL_TASK | true   | 100                           | 474                      |
+------+-------------+--------------------------+--------+-------------------------------+--------------------------+
2 rows in set (0.03 sec)
```

There're 2 exceptions, the first one is the same as issue description above:
```
[ERROR] 2022-03-21 17:40:06.978 [ShardingSphere-Command-0] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
java.lang.NullPointerException: null
	at org.apache.shardingsphere.sharding.merge.dql.groupby.GroupByMemoryMergedResult.getValueCaseSensitiveFromTables(GroupByMemoryMergedResult.java:134)
	at org.apache.shardingsphere.sharding.merge.dql.groupby.GroupByMemoryMergedResult.getValueCaseSensitive(GroupByMemoryMergedResult.java:124)
	at org.apache.shardingsphere.sharding.merge.dql.groupby.GroupByMemoryMergedResult.init(GroupByMemoryMergedResult.java:73)
	at org.apache.shardingsphere.sharding.merge.dql.groupby.GroupByMemoryMergedResult.init(GroupByMemoryMergedResult.java:53)
	at org.apache.shardingsphere.infra.merge.result.impl.memory.MemoryMergedResult.<init>(MemoryMergedResult.java:51)
	at org.apache.shardingsphere.sharding.merge.dql.groupby.GroupByMemoryMergedResult.<init>(GroupByMemoryMergedResult.java:56)
	at org.apache.shardingsphere.sharding.merge.dql.ShardingDQLResultMerger.getGroupByMergedResult(ShardingDQLResultMerger.java:112)
	at org.apache.shardingsphere.sharding.merge.dql.ShardingDQLResultMerger.build(ShardingDQLResultMerger.java:82)
	at org.apache.shardingsphere.sharding.merge.dql.ShardingDQLResultMerger.merge(ShardingDQLResultMerger.java:63)
	at org.apache.shardingsphere.infra.merge.MergeEngine.executeMerge(MergeEngine.java:90)
	at org.apache.shardingsphere.infra.merge.MergeEngine.merge(MergeEngine.java:80)
	at org.apache.shardingsphere.proxy.backend.communication.DatabaseCommunicationEngine.mergeQuery(DatabaseCommunicationEngine.java:159)
	at org.apache.shardingsphere.proxy.backend.communication.DatabaseCommunicationEngine.processExecuteQuery(DatabaseCommunicationEngine.java:113)
	at org.apache.shardingsphere.proxy.backend.communication.jdbc.JDBCDatabaseCommunicationEngine.execute(JDBCDatabaseCommunicationEngine.java:134)
	at org.apache.shardingsphere.proxy.backend.communication.jdbc.JDBCDatabaseCommunicationEngine.execute(JDBCDatabaseCommunicationEngine.java:70)
	at org.apache.shardingsphere.proxy.backend.text.data.impl.SchemaAssignedDatabaseBackendHandler.execute(SchemaAssignedDatabaseBackendHandler.java:55)
	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.text.query.MySQLComQueryPacketExecutor.execute(MySQLComQueryPacketExecutor.java:67)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:96)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:69)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

The second one occured on data consistency check stage:
```
[ERROR] 2022-03-21 17:42:00.271 [_finished_check_Worker-1] o.a.s.d.p.core.job.FinishedCheckJob - scaling job 0130317c30317c3054317c7465737431 finish check failed!
org.apache.shardingsphere.data.pipeline.core.exception.PipelineDataConsistencyCheckFailedException: could not get metadata for table 't_order'
	at org.apache.shardingsphere.data.pipeline.core.check.consistency.DataConsistencyCheckerImpl.lambda$checkRecordsContent$2(DataConsistencyCheckerImpl.java:167)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1082)
	at org.apache.shardingsphere.data.pipeline.core.check.consistency.DataConsistencyCheckerImpl.checkRecordsContent(DataConsistencyCheckerImpl.java:164)
	at org.apache.shardingsphere.data.pipeline.core.api.impl.RuleAlteredJobAPIImpl.dataConsistencyCheck0(RuleAlteredJobAPIImpl.java:263)
	at org.apache.shardingsphere.data.pipeline.core.api.impl.RuleAlteredJobAPIImpl.dataConsistencyCheck(RuleAlteredJobAPIImpl.java:245)
	at org.apache.shardingsphere.data.pipeline.core.job.FinishedCheckJob.dataConsistencyCheck(FinishedCheckJob.java:116)
	at org.apache.shardingsphere.data.pipeline.core.job.FinishedCheckJob.execute(FinishedCheckJob.java:81)
	at org.apache.shardingsphere.elasticjob.simple.executor.SimpleJobExecutor.process(SimpleJobExecutor.java:33)
	at org.apache.shardingsphere.elasticjob.simple.executor.SimpleJobExecutor.process(SimpleJobExecutor.java:29)
	at org.apache.shardingsphere.elasticjob.executor.ElasticJobExecutor.process(ElasticJobExecutor.java:172)
	at org.apache.shardingsphere.elasticjob.executor.ElasticJobExecutor.process(ElasticJobExecutor.java:141)
	at org.apache.shardingsphere.elasticjob.executor.ElasticJobExecutor.execute(ElasticJobExecutor.java:124)
	at org.apache.shardingsphere.elasticjob.executor.ElasticJobExecutor.execute(ElasticJobExecutor.java:100)
	at org.apache.shardingsphere.elasticjob.lite.internal.schedule.LiteJob.execute(LiteJob.java:35)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
```

**Comment by sandynz** [comment]
And need to add `@NoArgsConstructor` for `OpenGaussPipelineSQLBuilder` for now.

**Comment by sandynz** [comment]
Verified on latest master branch, on plain schema (common t_order, t_order_item), it works now. Seems it's fixed by #16247.

But for BenchmarkSQL, since there's a new binding table DistSQL issue, it could not be verified now, it'll be verified later.

**Comment by sandynz** [comment]
It works for BenchmarkSQL now, on master branch, commit 140f14664a14f14902b613561c32eb7b30540058. Binding table is ignored for now.
